src/main/scala/org/ergoplatform/nodeView/mempool/ErgoMemPool.scala

```diff
@@ -51,6 +51,16 @@ class ErgoMemPool private[mempool](pool: OrderedTxPool)(implicit settings: ErgoS
     new ErgoMemPool(pool.filter(condition))
   }
 
+  override def digest(size: Int): Seq[ErgoTransaction] = {
+    val txs = pool.orderedTransactions.values.toArray
+    val (digest, _, _) = (0 to size).foldLeft(Seq.empty[ErgoTransaction], txs, txs.length) {
```

Comment:
I suggest to implement helper function which generated random subset of indexes in an 0 to size range.
Maybe such function already exists
Then you can use simple array indexing to extract txs.